src/coreml.cs

```diff
@@ -0,0 +1,273 @@
+﻿//
+// CoreML C# bindings
+//
+// Authors:
+//	Alex Soto  <alexsoto@microsoft.com>
+//
+// Copyright 2017 Xamarin Inc. All rights reserved.
+//
+
+#if XAMCORE_2_0
+
+using System;
+using XamCore.ObjCRuntime;
+using XamCore.CoreFoundation;
+using XamCore.Foundation;
+
+#if !WATCH
+using XamCore.CoreVideo;
+#endif
+
+namespace XamCore.CoreML {
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Native]
+	public enum MLFeatureType : nint {
+		Invalid = 0,
+		Int64 = 1,
+		Double = 2,
+		String = 3,
+		Image = 4,
+		MultiArray = 5,
+		Dictionary = 6
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[ErrorDomain ("MLModelErrorDomain")]
+	[Native]
+	public enum MLModelError : nint {
+		Generic = 0,
+		FeatureType = 1,
+		DescriptionMismatch = 2,
+		Io = 3
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Native]
+	public enum MLMultiArrayDataType : nint {
+		Double = 65536 | 64,
+		Float32 = 65536 | 32,
+		Int32 = 131072 | 32
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLDictionaryFeatureProvider : MLFeatureProvider {
+
+		[Export ("dictionary")]
+		NSDictionary<NSString, MLFeatureValue> Dictionary { get; }
+
+		[Export ("initWithDictionary:error:")]
+		IntPtr Constructor (NSDictionary<NSString, NSObject> dictionary, out NSError error);
+
+		[Export ("objectForKeyedSubscript:")]
+		[return: NullAllowed]
+		MLFeatureValue ObjectForKeyedSubscript (string featureName);
```

Comment:
that's already exposed by `this [string]` right ?
I don't mind having a method (easier to discover) but it should be named like one ;-) as this one lack a verb/action
e.g. `GetValue (string)`
The alternative is to mark it as `[Internal]`